modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb

```diff
@@ -41,14 +41,21 @@ def bgs_service
                             external_key: target_veteran.participant_id)
         end
 
-        def generate_show_output(bgs_claim:, lighthouse_claim:)
+        def evss_docs_service
+          EVSS::DocumentsService.new(auth_headers)
+        end
+
+        def generate_show_output(bgs_claim:, lighthouse_claim:) # rubocop:disable Metrics/MethodLength
```

Comment:
I think you probably want to follow the pattern that you already established when you did the work to add the `errors` attribute.

Something more like this ::

```
def generate_show_output(bgs_claim:, lighthouse_claim:)
  if lighthouse_claim.present? && bgs_claim.present?
    bgs_details = bgs_claim[:benefit_claim_details_dto]
    structure = build_claim_structure(
      data: bgs_details,
      lighthouse_id: lighthouse_claim.id,
      upstream_id: bgs_details[:benefit_claim_id]
    )
  elsif lighthouse_claim.present? && bgs_claim.blank?
    structure = {
      lighthouse_id: lighthouse_claim.id,
      type: lighthouse_claim.claim_type,
      status: lighthouse_claim.status.capitalize
    }
  else
    bgs_details = bgs_claim[:benefit_claim_details_dto]
    structure = build_claim_structure(data: bgs_details,
                                      lighthouse_id: nil,
                                      upstream_id: bgs_details[:benefit_claim_id])
  end
  structure.merge!(errors: get_errors(lighthouse_claim))
  structure.merge!(supporting_documents: build_supporting_docs(bgs_claim: bgs_claim)
end

def build_supporting_docs(bgs_claim:)
  return [] unless bgs_claim

  evss_docs = evss_docs_service.get_claim_documents(bgs_claim[:benefit_claim_details_dto][:benefit_claim_id]).body

  return [] unless evss_docs

  evss_docs['documents'].map do |doc|
    {
      documentId: doc['document_id'],
      documentTypeLabel: doc['document_type_label'],
      originalFileName: doc['original_file_name'],
      trackedItemId: doc['tracked_item_id'],
      uploadDate: doc['upload_date']
    }
  end
end
```